middleware.js

```diff
@@ -132,54 +136,74 @@ module.exports = function(options) {
           error(err);
 
           return next(err);
-        } else {
-          data = result.css;
+        }
 
-          if (debug) {
-            log('render', options.response ? '<response>' : sassPath);
+        data = result.css;
+
+        if (debug) {
+          log('render', options.response ? '<response>' : sassPath);
+
+          if (sourceMap) {
+            log('render', this.options.sourceMap);
+          }
+        }
+        imports[sassPath] = result.stats.includedFiles;
 
-            if (sourceMap) {
-              log('render', this.options.sourceMap);
+        var cssDone = true;
+        var sourceMapDone = true;
+
+        function doneWriting() {
+          if (cssDone && sourceMapDone) {
+            if (options.response === false) {
+              next(sassMiddlewareError);
```

Comment:
Actually `node-sass` returns null when there is no error. That shouldn't be a problem.